packages/gatsby/src/utils/api-runner-node.js

```diff
@@ -115,10 +115,128 @@ const deferActions = actions => {
   return deferred
 }
 
-const getLocalReporter = (activity, reporter) =>
-  activity
-    ? { ...reporter, panicOnBuild: activity.panicOnBuild.bind(activity) }
-    : reporter
+/**
+ * Create a local reporter
+ * Used to override reporter methods with activity methods
+ */
+function getLocalReporter({ activity, reporter }) {
+  // If we have an activity, bind panicOnBuild to the activities method to
+  // join them
+  if (activity) {
+    return { ...reporter, panicOnBuild: activity.panicOnBuild.bind(activity) }
+  }
+
+  return reporter
+}
+
+function extendErrorIdWithPluginName(pluginName, errorMeta) {
+  if (typeof errorMeta === `object`) {
+    const id = errorMeta && errorMeta[`id`]
+
+    if (id) {
+      errorMeta[`id`] = `${pluginName}_${id}`
+    }
+  }
```

Comment:
Nit, especially since the check is done later anyways, but usually if you don't want `null` you add it to the same check to be clear

Was there a reason for using ``errorMeta[`id`]`` over a regular property access?

```suggestion
  if (typeof errorMeta === `object` && errorMeta) {
    const id = errorMeta.id

    if (id) {
      errorMeta.id = `${pluginName}_${id}`
    }
  }
```

Reply:
You could even do a little more lazily :shrug: Unless reading `id` is very expensive for some reason.

```suggestion
  if (errorMeta?.id) {
      errorMeta.id = `${pluginName}_${errorMeta.id}`
  }
```